tests/extend_gpio/main.c

```diff
@@ -0,0 +1,191 @@
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ *
+ * @file
+ * @brief       GPIO extension test routine
+ *
+ * @author      Matthew Blue <matthew.blue.neuro@gmail.com>
+ * @}
+ */
+
+#include <stdio.h>
+
+#include "extend/gpio.h"
+#include "periph/gpio.h"
+
+/* GPIO extension test functions */
+int test_gpio_init(void *dev, gpio_t pin, gpio_mode_t mode);
+int test_gpio_init_int(void *dev, gpio_t pin, gpio_mode_t mode,
+                       gpio_flank_t flank, gpio_cb_t cb, void *arg);
+void test_gpio_irq_enable(void *dev, gpio_t pin);
+void test_gpio_irq_disable(void *dev, gpio_t pin);
+int test_gpio_read(void *dev, gpio_t pin);
+void test_gpio_set(void *dev, gpio_t pin);
+void test_gpio_clear(void *dev, gpio_t pin);
+void test_gpio_toggle(void *dev, gpio_t pin);
+void test_gpio_write(void *dev, gpio_t pin, int value);
+
+/* GPIO extension test driver */
+const gpio_ext_driver_t tests_extend_gpio_driver = {
+    .init = test_gpio_init,
+    .init_int = test_gpio_init_int,
+    .irq_enable = test_gpio_irq_enable,
+    .irq_disable = test_gpio_irq_disable,
+    .read = test_gpio_read,
+    .set = test_gpio_set,
+    .clear = test_gpio_clear,
+    .toggle = test_gpio_toggle,
+    .write = test_gpio_write,
+};
+
+int test_gpio_init(void *dev, gpio_t pin, gpio_mode_t mode)
+{
+    (void)mode;
+
+    printf("init on dev 0x%x with pin %u\n",
+        (uint16_t)(uintptr_t)dev, (unsigned int)pin);
+
+    return 0;
+}
+
+int test_gpio_init_int(void *dev, gpio_t pin, gpio_mode_t mode,
+                       gpio_flank_t flank, gpio_cb_t cb, void *arg)
+{
+    (void)mode;
+    (void)flank;
+    (void)cb;
+    (void)arg;
+
+    printf("init_int on dev 0x%x with pin %u\n",
+        (uint16_t)(uintptr_t)dev, (unsigned int)pin);
+
+    return 0;
+}
+
+void test_gpio_irq_enable(void *dev, gpio_t pin)
+{
+    printf("irq_enable on dev 0x%x with pin %u\n",
+        (uint16_t)(uintptr_t)dev, (unsigned int)pin);
+}
+
+void test_gpio_irq_disable(void *dev, gpio_t pin)
+{
+    printf("irq_disable on dev 0x%x with pin %u\n",
+        (uint16_t)(uintptr_t)dev, (unsigned int)pin);
+}
+
+int test_gpio_read(void *dev, gpio_t pin)
+{
+    printf("read on dev 0x%x with pin %u\n",
+        (uint16_t)(uintptr_t)dev, (unsigned int)pin);
+
+    return 0;
+}
+
+void test_gpio_set(void *dev, gpio_t pin)
+{
+    printf("set on dev 0x%x with pin %u\n",
+        (uint16_t)(uintptr_t)dev, (unsigned int)pin);
+}
+
+void test_gpio_clear(void *dev, gpio_t pin)
+{
+    printf("clear on dev 0x%x with pin %u\n",
+        (uint16_t)(uintptr_t)dev, (unsigned int)pin);
+}
+
+void test_gpio_toggle(void *dev, gpio_t pin)
+{
+    printf("toggle on dev 0x%x with pin %u\n",
+        (uint16_t)(uintptr_t)dev, (unsigned int)pin);
+}
+
+void test_gpio_write(void *dev, gpio_t pin, int value)
+{
+    (void)value;
+
+    printf("write on dev 0x%x with pin %u\n",
+        (uint16_t)(uintptr_t)dev, (unsigned int)pin);
+}
+
+int main(void)
+{
+    uint16_t num;
```

Comment:
I started to adopt ESP8266 and ESP32 MCUs for your GPIO ext interface. This sould probably be 
```
 gpio_t num;
```
Otherwise you get the compile error below on platforms that use
```
typedef unsigned int gpio_t;
```
from `drivers/include/perihp/gpio.h` and where `unsigned int` is 32 bit, e.g., ESP8266 and ESP32.
```
/data/riotbuild/tests/extend_gpio/main.c:180:5: error: comparison is always true due to limited range of data type [-Werror=type-limits]
     for (num = 0; num <= GPIO_EXT_PIN_MASK; num++) {
```